packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react'
+import { IconBlockchain } from '@blockchain-com/icons'
+import styled, { keyframes } from 'styled-components'
+
+import { Flex } from 'components/Flex'
+
+import { Confirmation } from './Confirmation'
+import { Connected } from './Connected'
+import { Connecting } from './Connecting'
+
+const ConnectWrapper = styled(Flex)`
+  height: 100%;
+  justify-content: center;
+  align-items: center;
+`
+const showingFrames = keyframes`
+  from { opacity: 0; transform: scale(0.1) }
+  to { opacity: 1; transform: scale(1)}
+`
+const BlockchainIcon = styled(IconBlockchain)`
+  animation: ${showingFrames} 2s ease-in-out;
+  color: ${(props) => props.theme.white};
+`
+
+export type ConnectStep = 'InitialScreen' | 'Confirmation' | 'Connectiing' | 'Connected'
```

Comment:
I would use enum here. In the code below you use plain strings to set step but with enum you could use named fields